src/include/txn_inline.h

```diff
@@ -1132,14 +1125,24 @@ __wt_txn_read(
             if (cbt->upd_value->skip_buf) {
                 cbt->upd_value->buf.data = NULL;
                 cbt->upd_value->buf.size = 0;
+            } else if (ovfl_retry && is_ovfl_rm) {
+                ovfl_retry = false;
+                WT_STAT_CONN_DATA_INCR(session, txn_read_race_overflow_remove);
```

Comment:
Thanks for the stats. In case we end up looping indefinitely, the stat will be useful.

Reply:
Actually my bad, we don't retry more than once.

Reply:
The ovfl_retry flag can also protect us from going into an infinite loop.

Reply:
But nothing guarantees us that we won't read again before checkpoint reconciliation is over?
```
                 * Retry the
                 * read as the value should now be appended to the update chain by checkpoint
                 * reconciliation.
```
What if the value is still not happened to the update chain and we read again?

Reply:
In reconciliation, we always first append the value to the update chain then free the overflow blocks at the end. So if we see the block is freed then we must have already appended the value to the update chain.

Reply:
I don't understand clearly why we need to check and retry later after checking all the conditions of visibility instead of doing the retry whenever we found that the returned value is an overflow remove as we shouldn't expect this to be returned.

Reply:
@kommiharibabu We only need to retry if we want to return on page value and it is overflow removed. We read the onpage value before we actually decide whether we need to return it or not. If the onpage value is anyway not visible, no need to retry again.

Reply:
understood. Thanks.